packages/hub/tasks/create-profile.ts

```diff
@@ -0,0 +1,77 @@
+import { inject } from '@cardstack/di';
+import { query } from '@cardstack/hub/queries';
+import { service } from '@cardstack/hub/services';
+import { encodeDID } from '@cardstack/did-resolver';
+import * as Sentry from '@sentry/node';
+import config from 'config';
+import logger from '@cardstack/logger';
+
+const log = logger('hub/worker');
+
+const network = config.get('web3.layer2Network') as string;
+
+const merchantCreationsQuery = `
+  query($txn: String!) {
+    transaction(id: $txn) {
+      merchantCreations {
+        merchant {
+          id
+        }
+        merchantSafe {
+          id
+        }
+      }
+    }
+  }
+`;
+
+export default class CreateProfile {
+  cardpay = inject('cardpay');
+  jobTickets = query('job-tickets', { as: 'jobTickets' });
+  merchantInfoQueries = query('merchant-info', {
+    as: 'merchantInfoQueries',
+  });
+  relay = service('relay');
+  web3 = inject('web3-http', { as: 'web3' });
+  workerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform({
+    'job-ticket-id': jobTicketId,
+    'merchant-info-id': merchantInfoId,
+  }: {
+    'job-ticket-id': string;
+    'merchant-info-id': string;
+  }) {
+    try {
+      let merchantInfo = (await this.merchantInfoQueries.fetch({ id: merchantInfoId }))[0];
+      let did = encodeDID({ type: 'MerchantInfo', uniqueId: merchantInfoId });
+
+      let profileRegistrationTxHash = await this.relay.registerProfile(merchantInfo.ownerAddress, did);
+
+      await this.cardpay.waitForTransactionConsistency(this.web3.getInstance(), profileRegistrationTxHash);
+
+      let merchantCreationsSubgraphResult = await this.cardpay.gqlQuery(network, merchantCreationsQuery, {
+        txn: profileRegistrationTxHash,
+      });
```

Comment:
Is there a risk of this graph query failing and causing the job ticket to fail but, the profile was actually already created? In that situation will retrying the job ticket succeed?

Reply:
I’m unsure. The issue specifies that this job should not automatically be retried. But the next issue I’m going to work on lets Card Wallet retry a job from its ticket id… but that would indeed create a second merchant safe, undesirable 🤔

Reply:
Yeah, if a txn hash exists, the system needs to assume that the safe also exists, despite the inability to resolve the merchant safe id if the graph node is unavailable.